test/api_s3.dart

```diff
@@ -0,0 +1,106 @@
+import 'dart:math';
+import 'dart:io';
+import 'package:test/test.dart';
+import 'package:pilot_s3/api/api_s3.dart';
+import 'package:crypto/crypto.dart' as crypto;
+
+String generateRandomString(int len) {
+  var r = Random();
+  const chars =
+      'AaBbCcDdEeFfGgHhIiJjKkLlMmNnOoPpQqRrSsTtUuVvWwXxYyZz1234567890';
+  return List.generate(len, (index) => chars[r.nextInt(chars.length)]).join();
+}
+
+void main() {
+  String bucketName = Platform.environment['TEST_S3_BUCKET'] ?? '';
+
+  ApiS3 api = ApiS3(
+      endPoint: Platform.environment['TEST_S3_ENDPOINT'] ?? '',
+      accessKey: Platform.environment['TEST_S3_ACCESS_KEY'] ?? '',
+      secretKey: Platform.environment['TEST_S3_SECRET_KEY'] ?? '',
+      defaultBucket: bucketName);
+
+  var objectPath = 'test${generateRandomString(10)}.txt';
+  var filePath = 'test/$objectPath';
+  var filePathCheck = '$filePath.check';
+
+  var file = File(filePath);
+  var fileCheck = File(filePathCheck);
+
+  setUp(() async {
+    // TODO: up docker minio s3
+  });
+
+  tearDown(() async {
+    if (await file.exists()) {
+      await file.delete();
+    }
+    if (await fileCheck.exists()) {
+      await fileCheck.delete();
+    }
+  });
+
+  group('api: s3', () {
+    test('get buckets', () async {
+      var buckets = await api.getBuckets();
+
+      expect(buckets.length, equals(1), reason: 'buckets list not empty');
+      expect(buckets[0].name, equals(bucketName),
+          reason: 'bucket with current name exist');
+    });
+
+    test('get buckets count', () async {
+      var bucketsCount = await api.getBucketsCount();
+
+      expect(bucketsCount, equals(1), reason: 'buckets count not zero');
+    });
+
+    test('get buckets count', () async {
+      var bucketsCount = await api.getBucketsCount();
+
+      expect(bucketsCount, equals(1), reason: 'bucket exist');
+    });
+
+    test('upload - download - delete object', () async {
+      await api.deleteObject(objectPath);
```

Comment:
mb separate it to three independent tests?

Reply:
I thought about it, but before download something, we must upload something
And separated test after it useless